test/Microsoft.Azure.SignalR.E2ETests/Management/ServiceHubContextE2EFacts.cs

```diff
@@ -294,9 +294,57 @@ internal async Task ConnectionJoinLeaveGroupTest(ServiceTransportType serviceTra
             }
         }
 
+        [ConditionalFact]
+        [SkipIfConnectionStringNotPresent]
+        internal async Task WithEndpointsTest()
+        {
+            using (StartVerifiableLog(out var loggerFactory, LogLevel.Debug))
+            {
+                var services = new ServiceCollection().AddSignalRServiceManager().AddSingleton(TestConfiguration.Instance.Configuration).AddSingleton(loggerFactory);
+                var serviceProvider = services.AddSingleton<IReadOnlyCollection<ServiceDescriptor>>(services.ToList()).BuildServiceProvider();
+                var hubContext = await serviceProvider.GetRequiredService<IServiceManager>().CreateHubContextAsync(HubName);
+                var endpointManager = serviceProvider.GetRequiredService<IServiceEndpointManager>();
+                var endpoints = endpointManager.GetEndpoints(HubName).ToArray<ServiceEndpoint>();
+                if (endpoints.Length < 2)
+                {
+                    throw new InvalidOperationException("This test need at lease two Azure SignalR instance endpoints.");
+                };
+                var connections = endpoints.Select(endpoint =>
+                {
+                    var provider = endpointManager.GetEndpointProvider(endpoint);
+                    var clientEndpoint = provider.GetClientEndpoint(HubName, null, null);
+                    var token = provider.GenerateClientAccessTokenAsync(HubName).Result;
+                    return CreateHubConnection(clientEndpoint, token);
+                }).ToArray();
+                using var cancellationTokenSource = new CancellationTokenSource();
+                await Task.WhenAll(connections.Select(conn => conn.StartAsync()));
+                HandleHubConnection(connections, cancellationTokenSource);
+                var receivedFlags = new bool[endpoints.Length];
+                for (var i = 0; i < endpoints.Length; i++)
+                {
+                    var j = i;
+                    connections[j].On(MethodName, (string message) =>
+                    {
+                        receivedFlags[j] = true;
+                    });
+                }
+
+                var subHubContext = (hubContext as IInternalServiceHubContext).WithEndpoints(endpoints.Take(1));
```

Comment:
looks like `WithEnpoint()` equals to build a new serviceManager then create hub context.
Since it's internal in asrs function extension, why not simply do that?

Reply:
It is not equal to build a new ServiceManager, because it doesn't create new connections.

Reply:
Then you limit the endpoints has to be the subset of the endpoints.
For the old version, in binding attribute, cx can set any connection string key no matter which one is for glabal usage.
I think we can keep it too.